### ISSUE TITLE
'IndexError: tuple index out of range' when use pd.read_pickle

### ISSUE THREAD

**Issue by auwsom** [quantopian/zipline#2069]
Dear Zipline Maintainers,

Before I tell you about my issue, let me describe my environment:

# Environment

* Operating System: (Windows Version or `$ uname --all`)
Windows 10
* Python Version: `$ python --version`
Anaconda 3 install 3.6.3, conda update --all
pip install zipline --update
* Python Bitness: `$ python -c 'import math, sys;print(int(math.log(sys.maxsize + 1, 2) + 1))'`
64 bit
* How did you install Zipline: (`pip`, `conda`, or `other (please explain)`)
pip in Anaconda terminal window
* Python packages: `$ pip freeze` or `$ conda list`

Now that you know a little about me, let me tell you about the issue I am
having:

# Description of Issue

* What did you expect to happen?
Running basic example from zipline tutorial. After finally getting data download, after the Google/Yahoo/IEX fix.

* What happened instead?
'IndexError: tuple index out of range' when use pd.read_pickle

Here is how you can reproduce this issue on your machine:

## Reproduction Steps

%load_ext zipline
%%zipline --start 2000-1-1 --end 2014-1-1 -o buyapple_out.pickle 
(where does this go by default, couldn't find anywhere, had to use -o c:\buyapple_out.pickle and copy to Jupyter folder)

import pandas as pd
perf = pd.read_pickle('buyapple_out.pickle') # read in perf DataFrame
...

## What steps have you taken to resolve this already?
Found issue about conflict between pandas and pytz (see link below) and cross-referencing for others searching this issue.
...

# Anything else?
zipline issue:
https://github.com/quantopian/zipline/issues/654
pandas issue:
https://github.com/pandas-dev/pandas/issues/6871

here's the full trace:
it looks like the first error is 'TypeError: Cannot convert input to Timestamp'
I get the same error with pickle.load(), so I'm assuming the zipline output is an incorrect pickle format:
import pickle
with open('buyapple_out.pickle', 'rb') as f:
      panel = pickle.load(f)    

TypeError: Cannot convert input to Timestamp

---
TypeError                                 Traceback (most recent call last)
~\Anaconda3\lib\site-packages\pandas\io\pickle.py in try_read(path, encoding)
     51             with open(path, 'rb') as fh:
---> 52                 return pkl.load(fh)
     53         except Exception:

pandas/tslib.pyx in pandas.tslib.Timestamp.__new__ (pandas\tslib.c:9203)()

pandas/tslib.pyx in pandas.tslib.convert_to_tsobject (pandas\tslib.c:25854)()

TypeError: Cannot convert input to Timestamp

During handling of the above exception, another exception occurred:

IndexError                                Traceback (most recent call last)
~\Anaconda3\lib\site-packages\pandas\io\pickle.py in try_read(path, encoding)
     56                 with open(path, 'rb') as fh:
---> 57                     return pc.load(fh, encoding=encoding, compat=False)
     58 

~\Anaconda3\lib\site-packages\pandas\compat\pickle_compat.py in load(fh, encoding, compat, is_verbose)
    116 
--> 117         return up.load()
    118     except:

~\Anaconda3\lib\pickle.py in load(self)
   1049                 assert isinstance(key, bytes_types)
-> 1050                 dispatch[key[0]](self)
   1051         except _Stop as stopinst:

~\Anaconda3\lib\site-packages\pandas\compat\pickle_compat.py in load_reduce(self)
     16 
---> 17     if type(args[0]) is type:
     18         n = args[0].__name__

IndexError: tuple index out of range

During handling of the above exception, another exception occurred:

IndexError                                Traceback (most recent call last)
~\Anaconda3\lib\site-packages\pandas\io\pickle.py in read_pickle(path)
     64     try:
---> 65         return try_read(path)
     66     except:

~\Anaconda3\lib\site-packages\pandas\io\pickle.py in try_read(path, encoding)
     61                 with open(path, 'rb') as fh:
---> 62                     return pc.load(fh, encoding=encoding, compat=True)
     63 

~\Anaconda3\lib\site-packages\pandas\compat\pickle_compat.py in load(fh, encoding, compat, is_verbose)
    116 
--> 117         return up.load()
    118     except:

~\Anaconda3\lib\pickle.py in load(self)
   1049                 assert isinstance(key, bytes_types)
-> 1050                 dispatch[key[0]](self)
   1051         except _Stop as stopinst:

~\Anaconda3\lib\site-packages\pandas\compat\pickle_compat.py in load_reduce(self)
     16 
---> 17     if type(args[0]) is type:
     18         n = args[0].__name__

IndexError: tuple index out of range

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
~\Anaconda3\lib\site-packages\pandas\io\pickle.py in try_read(path, encoding)
     51             with open(path, 'rb') as fh:
---> 52                 return pkl.load(fh)
     53         except Exception:

pandas/tslib.pyx in pandas.tslib.Timestamp.__new__ (pandas\tslib.c:9203)()

pandas/tslib.pyx in pandas.tslib.convert_to_tsobject (pandas\tslib.c:25854)()

TypeError: Cannot convert input to Timestamp

During handling of the above exception, another exception occurred:

IndexError                                Traceback (most recent call last)
~\Anaconda3\lib\site-packages\pandas\io\pickle.py in try_read(path, encoding)
     56                 with open(path, 'rb') as fh:
---> 57                     return pc.load(fh, encoding=encoding, compat=False)
     58 

~\Anaconda3\lib\site-packages\pandas\compat\pickle_compat.py in load(fh, encoding, compat, is_verbose)
    116 
--> 117         return up.load()
    118     except:

~\Anaconda3\lib\pickle.py in load(self)
   1049                 assert isinstance(key, bytes_types)
-> 1050                 dispatch[key[0]](self)
   1051         except _Stop as stopinst:

~\Anaconda3\lib\site-packages\pandas\compat\pickle_compat.py in load_reduce(self)
     16 
---> 17     if type(args[0]) is type:
     18         n = args[0].__name__

IndexError: tuple index out of range

During handling of the above exception, another exception occurred:

IndexError                                Traceback (most recent call last)
<ipython-input-3-f25ea3a5b421> in <module>()
     14 # with open('buyapple_out.pickle', 'rb') as f:
     15 #       panel = pickle.load(f)
---> 16 perf = pd.read_pickle('buyapple_out.pickle') # read in perf DataFrame
     17 perf.head()
     18 

~\Anaconda3\lib\site-packages\pandas\io\pickle.py in read_pickle(path)
     66     except:
     67         if PY3:
---> 68             return try_read(path, encoding='latin1')
     69         raise
     70 

~\Anaconda3\lib\site-packages\pandas\io\pickle.py in try_read(path, encoding)
     60             except:
     61                 with open(path, 'rb') as fh:
---> 62                     return pc.load(fh, encoding=encoding, compat=True)
     63 
     64     try:

~\Anaconda3\lib\site-packages\pandas\compat\pickle_compat.py in load(fh, encoding, compat, is_verbose)
    115         up.is_verbose = is_verbose
    116 
--> 117         return up.load()
    118     except:
    119         raise

~\Anaconda3\lib\pickle.py in load(self)
   1048                     raise EOFError
   1049                 assert isinstance(key, bytes_types)
-> 1050                 dispatch[key[0]](self)
   1051         except _Stop as stopinst:
   1052             return stopinst.value

~\Anaconda3\lib\site-packages\pandas\compat\pickle_compat.py in load_reduce(self)
     15     func = stack[-1]
     16 
---> 17     if type(args[0]) is type:
     18         n = args[0].__name__
     19 

IndexError: tuple index out of range

**Comment by freddiev4** [comment]
Hi @auwsom so looks like there's a few things going on here:

- By default, the pickle file should go in the directory in which you're running your zipline backtest or Jupyter Notebook
- I'll try to reproduce this in a Python 3.6 env; we don't yet have support/packages for Python 3.6 so there may be pickle/pandas formatting issues related to that. Can you try running in a Py35 env?
- With the IEX fix, you should only be able to run a 5 year backtest (up to 5 years from current day) so you might want to modify the notebook code to reflect that (I'm going to update the docs for that)

**Comment by auwsom** [comment]
Thanks @freddiev4! I'll try to get that tested under 3.5 and update this comment. It could be the IEX response if I was trying an incorrect date range. Do you have a small example pickle file you could share so I can compare the formats and test a known working file? Also, yes, strange the output doesn't land in the Jupyter directory, I tried that with and without admin Anaconda terminal, and check everywhere I could think of. Another person had this problem and posted on SO here: https://stackoverflow.com/questions/38337802/zipline-where-are-the-pickle-files-saved/48047227#48047227

**Comment by freddiev4** [comment]
@auwsom here's a `.pickle` file I just made, sharing via a Dropbox link: https://www.dropbox.com/s/r6psvk3rc0kfrtz/buyapple_out.pickle?dl=0 let me know if that helps at all 🙂

**Comment by auwsom** [comment]
Yes, you're file loads fine. So I need to figure out if I can compare files (not all is in text format) or try identifying which part is not working. I think that's a direct approach. Thanks for the pickle file!

**Comment by freddiev4** [comment]
Is there any reason why you're testing Yahoo/Google Finance errors? As the latest zipline master uses a different source (IEX).

**Comment by auwsom** [comment]
I was trying to fix the loader myself before I gave up and then found the thread there was a changeover happening. Then I retested when I was providing information and found my code error. It would be nice to have a backup source and also to have more of the historical data, but I'm glad IEX is working for people.

np++ is struggling to compare, but it looks like there's a way to output with pickle using ascii (I still need to be able to input it). There's also a way to view the text of a .pickle file via command prompt, if it works on Windows.
https://stackoverflow.com/questions/13568790/is-there-a-way-to-view-cpickle-or-pickle-file-contents-without-loading-python-in

**Comment by auwsom** [comment]
Seems to be working now. I installed Anaconda 3.5, and had problems installing zipline because of statsmodels error with pip about vs studio... conda said it was up-to-date, but didn't find zipline in its repo locations. Repaired VS 2017. Tried Anaconda 3.6 again just to check, and it works now, except I have to use an Admin window of Anaconda to open Jupyter, because it had a permissions error with the pickle file, for some reason. So made a Junction from my %homepath%\Jupyter to %System32%, and it is happy :)
